draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -728,56 +728,81 @@ If any step fails, the Presentation is not valid and processing MUST be aborted.
+to Section 7.2 of [@!RFC7515]. The disclosures (both for issuance and presentation) SHOULD be included in the
+serialized JWS using the member name `disclosures` at the top-level of the JSON object (the same level as the `payload` member). The
 value of the `disclosures` member is an array of strings where each element is an individual Disclosure
 as described in (#creating_disclosures). The Issuer includes a Disclosure for each selectively
 disclosable claim of the SD-JWT payload, whereas the Holder includes only the Disclosures
-selected for the given presentation. Additionally, for presentation with a Key Binding, the Holder adds
-the key `kb_jwt` at the top-level of the serialized JWS with a string value containing the
-Key Binding JWT as described in (#kb-jwt).
+selected for the given presentation.
+
+Alternative methods for conveying the disclosures MAY be used (such as including them in a `disclosures`
+member of an outer JSON structure also containing the JSON Serialized SD-JWT) as dictated by a specific
+application or transport protocol. However, the details of such approaches fall outside the scope of this
+specification.
 
 Verification of the JWS JSON serialized SD-JWT follows the same rules defined in (#verification),
-except that the SD-JWT does not need to be split into component parts, but disclosures and (if applicable)
-a Key Binding JWT can be found in the respective members of the JSON object.
+except that the SD-JWT does not need to be split into component parts, the disclosures
+can be found in the respective member of the JSON object (or elsewhere), and Key Binding (if applicable)
+will be provided by means not specifically defined in this specification.
 
 Using a payload similar to that from [Example 1](#example-1), the following is a non-normative example of
 a JWS JSON serialized SD-JWT from an Issuer with all the respective Disclosures.
 
 <{{examples/json_serialization/sd_jwt_issuance.json}}
 
 Below is a non-normative example of a presentation of the JWS JSON serialized SD-JWT, where the Holder
-includes a Key Binding JWT and has selected to disclose `given_name`, `family_name`, and `address`.
+has selected to disclose `given_name`, `family_name`, and `address`.
 
 <{{examples/json_serialization/sd_jwt_presentation.json}}
 
 
+# Enveloping SD-JWTs {#enveloping}
+
+In some applications or transport protocols, it is desirable to encapsulate an SD-JWT into an outer JWT container. For example, an implementation may enclose multiple credentials and presentations, independent of their format, in a JWT to enable application-layer encryption during transport.
+
+For such use cases, a compact serialized SD-JWT SHOULD be included as a single string value and a JSON serialized SD-JWT SHOULD be included as a JSON object value. Key Binding MAY be achieved by signing the envelope JWT instead of including a separate Key Binding JWT.
```

Comment:
If W3C VC-jose-cose spec adopts kb-jwt appended using ~, I would be open to mandating that as a way to do KB with compact serialization (cc @OR13 )

Reply:
Brian reminded me that we wanted this optionality to envelope not only for w3c, but also for the cases when the same KB JWT is used for the multiple issuer signed JWTs